### DISCUSSION TITLE
Using a `columns` shortcode from an theme to export HTML columns (not the columns in tables)

### DISCUSSION THREAD

**Discussion by johanwk** [kaushalmodi/ox-hugo#550] [Q&A] [answered]
The geekdoc theme allows for content typeset in columns, see https://geekdocs.de/shortcodes/columns/. This uses a magic separator `<--->` to separate columns.

Now, a `<` naturally gets translated into `&lt;`, but the magic separator can be protected with `#hugo:`, and then the output is quite correct: 
```
#+begin_columns
   left column text here text here
#+hugo: <--->
   right column text here text here
#+end_columns
```

However, this doesn't seem to be the right org-mode solution, since there is already `#+begin_columns` to start columns, and `#+begin_column` for each column; also, at https://sachachua.com/blog/category/writing/ there's `#+begin_column50`, obviously indicating a 50% width (although I couldn't find any of this in the org manual). I think the following would be the right way to write multicolumn content:
```
#+begin_columns
#+begin_column
   left column text here text here
#+end_column
#+begin_column
   right column text here text here
#+end_column
#+end_columns
```
Would this be a good suggestion for support in ox-hugo?

**Comment by kaushalmodi** [comment]
Hello,

> The geekdoc theme allows for content typeset in columns, see https://geekdocs.de/shortcodes/columns/. This uses a magic separator `<--->` to separate columns.

One way to is define that special block such that Org mode exports the inside as raw Org, without doing any conversion to Markdown or HTML. But then, inside that special block, you will need to type the raw Markdown+HTML, just as if you were typing in a Markdown file. 

Example:

```org
#+begin_columns
## Left Content
Dolor sit, sumo unique argument um no ...

<---> <!-- magic separator, between columns -->

## Mid Content
Dolor sit, sumo unique argument um no ...

<---> <!-- magic separator, between columns -->

## Right Content
Dolor sit, sumo unique argument um no ...
#+end_columns
```

See https://ox-hugo.scripter.co/doc/org-special-blocks/#raw-org-contents.

> also, at https://sachachua.com/blog/category/writing/ there's #+begin_column50, obviously indicating a 50% width (although I couldn't find any of this in the org manual).

Sacha Chua has defined a custom special block type called "column50". See https://sachachua.com/dotemacs/#org7deeabb (that link is not a permalink so it might change in future. So search for column50 on that page). 

> I think the following would be the right way to write multicolumn content:

That's true. Your last suggestion looks the cleanest, and you should be able to do that by putting in some work in a custom Hugo shortcode called `column`. Do you want to give that a try?

**Comment by kaushalmodi** [comment]
Hint: See that shortcode definition: https://github.com/thegeeklab/hugo-geekdoc/blob/main/layouts/shortcodes/columns.html and extract the inside of that shortcode into a new shortcode.

**Comment by kaushalmodi** [comment]
Or.. copy that shortcode definition to your site to override it and then change the `<--->` split marker to anything else of your choice 😉

**Reply by johanwk** [reply]
But -- there's an advantage to using the org-mode format with begin_/end_column, for instance in editing -- one can fold columns, for starters. So much better for editing.

(I'll make an attempt :) )

**Reply by kaushalmodi** [reply]
Yes, you can have both. Do the nested special blocks and customize that shortcode.

**Reply by johanwk** [reply]
I'm afraid this is beyond my ability, since I know very little about both Hugo and HTML. :/

**Comment by kaushalmodi** [comment, marked as answer]
@johanwk OK, I won't usually do this, but this was fun :)

Check out the instructions on https://geekdoc-theme-columns-sc--hugo-mwe.netlify.app/posts/columns/ on how to customize geekdoc theme's `columns` shortcode and use in Org.

[source commit](https://gitlab.com/hugo-mwe/hugo-mwe/-/commit/5660fbbcd426639b394a157ead455e759691af94)

**Reply by johanwk** [reply]
Fantastic, this works just as expected. Looking at the file, it's even something I could steal from with a bit of confidence, next tïme the need appears :)
